translation/message_format.rst

```diff
@@ -0,0 +1,443 @@
+                    <trans-unit id="finish_place">
+                        <source>finish_place</source>
+                        <target>You finished {place, selectordinal, one {#st} two {#nd} few {#rd} other {#th}}!</target>
+                    </trans-unit>
+
+                    <!-- when only formatting the number as ordinal (like
+                         above), you can also use the `ordinal` function: -->
+                    <trans-unit id="finish_place">
+                        <source>finish_place</source>
+                        <target>You finished {place, ordinal}!</target>
+                    </trans-unit>
+                </body>
+            </file>
+        </xliff>
+
+    .. code-block:: php
+
+        // translations/messages.en.php
+        return [
+            'finish_place' => 'You finished {place, selectordinal,
+                one {#st}
+                two {#nd}
+                few {#rd}
+                other {#th}
+            }!',
+
+            // when only formatting the number as ordinal (like above), you can
+            // also use the `ordinal` function:
+            'finish_place' => 'You finished {place, ordinal}!',
+        ];
+
+.. code-block:: php
+
+    // prints "You finished 1st!"
+    echo $translator->trans('finish_place', ['place' => 1]);
+
+    // prints "You finished 9th!"
+    echo $translator->trans('finish_place', ['place' => 9]);
+
+    // prints "You finished 23rd!"
+    echo $translator->trans('finish_place', ['place' => 23]);
+
+The possible cases for this are also shown in Unicode's `Language Plural Rules`_ document.
+
+Date and Time
+~~~~~~~~~~~~~
+
+The date and time function allows you to format dates in the target locale
+using the :phpclass:`IntlDateFormatter`:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # translations/messages.en.yaml
+        published_at: 'Published at {publication_date, date} - {publication_date, time, short}'
+
+    .. code-block:: xml
+
+        <!-- translations/messages.en.xlf -->
+        <?xml version="1.0"?>
+        <xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2">
+            <file source-language="en" datatype="plaintext" original="file.ext">
+                <body>
+                    <trans-unit id="published_at">
+                        <source>published_at</source>
+                        <target>Published at {publication_date, date} - {publication_date, time, short}</target>
+                    </trans-unit>
+                </body>
+            </file>
+        </xliff>
+
+    .. code-block:: php
+
+        // translations/messages.en.php
+        return [
+            'published_at' => 'Published at {publication_date, date} - {publication_date, time, short}',
+        ];
+
+The "function statement" for the ``time`` and ``date`` functions can be one of
+short, medium, long or full, as documented on PHP.net.
+
+.. code-block:: php
+
+    // prints "Published at Jan 25, 2019 - 11:30 AM"
```

Comment:
Maybe puts prints next to or after the echo?

Reply:
Hmm, I don't have a good reason, but I propose to leave it like it is. I don't think it's too confusing and moving them all down is quite time consuming :) Please let me know if it was confusing for you!